lib/monitoring/logging-engine.js

```diff
@@ -0,0 +1,145 @@
+const LogLevel = {
+  OFF: 0,
+  TRACE: 1,
+  DEBUG: 2,
+  INFO: 3,
+  WARNING: 4,
+  ERROR: 5,
+  CRITICAL: 6
+};
+
+const logLevelAsText = [
+  'OFF',
+  'TRACE',
+  'DEBUG',
+  'INFO',
+  'WARNING',
+  'ERROR',
+  'CRITICAL'
+];
```

Comment:
I've been using `Symbol` for this purpose. E.g. 
```const SupervisionActions = {
  stop: Symbol('stop'),
  stopAll: Symbol('stopAll'),
  escalate: Symbol('escalate'),
  resume: Symbol('resume'),
  reset: Symbol('reset'),
  resetAll: Symbol('resetAll')
};```

Reply:
But I see that you're using the values in the LogLevel to define a natural ordering

Reply:
So ignore this